filesystems/blob/src/test/scala/cromwell/filesystems/blob/BlobPathBuilderSpec.scala

```diff
@@ -0,0 +1,40 @@
+package cromwell.filesystems.blob
+
+import com.azure.core.credential.AzureSasCredential
+import cromwell.filesystems.blob.BlobPathBuilder
+import org.scalatest.Ignore
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+import java.nio.file.Files
+
+@Ignore
+class BlobPathBuilderSpec extends AnyFlatSpec with Matchers{
+
+  it should "parse a URI into a path" in {
+    val testString = "https://storageAccount.blob.core.windows.net/container/path/to/file"
+    val evalPath = "/path/to/file"
+    val pathBuilder = new BlobPathBuilder(new AzureSasCredential("test"), "container", "storageAccount")
+    pathBuilder.validateBlobPath(testString) match {
+      case BlobPathBuilder.ValidBlobPath(path) => path should equal(evalPath)
+      case BlobPathBuilder.UnparsableBlobPath(errorMessage) => fail(errorMessage)
+    }
```

Comment:
I think this is a good example of the advantage of putting a method in the companion object, and having the class instance call it with generic arguments. That gives us the ability to test the logic directly using only the inputs it needs, rather than creating a `BlobPathBuilder` that we don't need (specifically, it lets us not worry about the token).

You could have: 
```
object BlobPathBuilder {
    def validateBlobPath(path: String, container: String, storageAccount: String): BlobPathValidation = {
        ...
    }
}
```
...then you could call it directly from both test code and the class instance method.